docs/01-app/03-api-reference/01-directives/use-cache-remote.mdx

```diff
@@ -323,16 +248,120 @@ async function innerPrivate() {
 
+### Basic example
+
+Cache product pricing that needs to be fetched at request time but can be shared across all users. Use [`cacheLife`](/docs/app/api-reference/functions/cacheLife#custom-cache-profiles) to set the cache lifetime of the price.
+
+```tsx filename="app/product/[id]/page.tsx" switcher
+import { Suspense } from 'react'
+import { connection } from 'next/server'
+import { cacheTag, cacheLife } from 'next/cache'
+
+export async function generateStaticParams() {
+  return [{ id: '1' }, { id: '2' }, { id: '3' }]
+}
+
+export default async function ProductPage({
+  params,
+}: {
+  params: Promise<{ id: string }>
+}) {
+  const { id } = await params
+
+  return (
+    <div>
+      <ProductDetails id={id} />
+      <Suspense fallback={<div>Loading price...</div>}>
+        <ProductPrice productId={id} />
+      </Suspense>
+    </div>
+  )
+}
+
+function ProductDetails({ id }: { id: string }) {
+  return <div>Product: {id}</div>
+}
+
+async function ProductPrice({ productId }: { productId: string }) {
+  // Calling connection() makes this component dynamic, preventing
+  // it from being included in the static shell. This ensures the price
+  // is always fetched at request time.
+  await connection()
+
+  // Now we can cache the price in a remote cache handler.
+  const price = await getProductPrice(productId)
+
+  return <div>Price: ${price}</div>
+}
+
+async function getProductPrice(productId: string) {
+  'use cache: remote'
+  cacheTag(`product-price-${productId}`)
+  cacheLife({ expire: 3600 }) // 1 hour
+
+  // This database query is cached and shared across all users
+  return db.products.getPrice(productId)
+}
+```
+
+```jsx filename="app/product/[id]/page.js" switcher
+import { Suspense } from 'react'
+import { connection } from 'next/server'
+import { cacheTag, cacheLife } from 'next/cache'
+
+export async function generateStaticParams() {
+  return [{ id: '1' }, { id: '2' }, { id: '3' }]
+}
+
+export default async function ProductPage({ params }) {
+  const { id } = await params
+
+  return (
+    <div>
+      <ProductDetails id={id} />
+      <Suspense fallback={<div>Loading price...</div>}>
+        <ProductPrice productId={id} />
+      </Suspense>
+    </div>
+  )
+}
+
+function ProductDetails({ id }) {
+  return <div>Product: {id}</div>
+}
+
+async function ProductPrice({ productId }) {
+  // Calling connection() makes this component dynamic, preventing
+  // it from being included in the static shell. This ensures the price
+  // is always fetched at request time.
+  await connection()
+
+  // Now we can cache the price in a remote cache handler.
+  const price = await getProductPrice(productId)
+
+  return <div>Price: ${price}</div>
+}
+
+async function getProductPrice(productId) {
+  'use cache: remote'
+  cacheTag(`product-price-${productId}`)
+  cacheLife({ expire: 3600 }) // 1 hour
+
+  // This database query is cached and shared across all users
+  return db.products.getPrice(productId)
+}
+```
+
```

Comment:
I think what makes this example confusing to me is that the params are assumed to be known (gSP) but the price data is held back via connection. If the cacheLife is 1 hour, why not include it in the shell? It will still be revalidated every hour, meaning the upstream db will be hit every hour, right? What's the benefit of using remote caching here?